core/arch/arm/plat-imx/config/imx6q.h

```diff
@@ -0,0 +1,267 @@
+ * bit[6:4]:2 - 3 cycle of read accesses latency
+ * bit[2:0]:2 - 3 cycle of setup latency
+ */
+#ifndef PL310_DATA_RAM_CTRL_INIT
+#define PL310_DATA_RAM_CTRL_INIT	0x00000222
+#endif
+
+/*
+ * PL310 Auxiliary Control Register
+ *
+ * I/Dcache prefetch enabled (bit29:28=2b11)
+ * NS can access interrupts (bit27=1)
+ * NS can lockown cache lines (bit26=1)
+ * Pseudo-random replacement policy (bit25=0)
+ * Force write allocated (default)
+ * Shared attribute internally ignored (bit22=1, bit13=0)
+ * Parity disabled (bit21=0)
+ * Event monitor disabled (bit20=0)
+ * Platform flavor specific way config (dual / quad):
+ * - 64kb way size (bit19:17=3b011)
+ * - 16-way associativity (bit16=1)
+ * Platform flavor specific way config (dual lite / solo):
+ * - 32kb way size (bit19:17=3b010)
+ * - no 16-way associativity (bit16=0)
+ * Store buffer device limitation enabled (bit11=1)
+ * Cacheable accesses have high prio (bit10=0)
+ * Full Line Zero (FLZ) disabled (bit0=0)
+ */
+#ifndef PL310_AUX_CTRL_INIT
+#if defined(CFG_MX6QP) || defined(CFG_MX6Q) || defined(CFG_MX6D)
+#define PL310_AUX_CTRL_INIT		0x3C470800
+#else
+#define PL310_AUX_CTRL_INIT		0x3C440800
+#endif
+#endif
+
+/*
+ * PL310 Prefetch Control Register
+ *
+ * Double linefill disabled (bit30=0)
+ * I/D prefetch enabled (bit29:28=2b11)
+ * Prefetch drop enabled (bit24=1)
+ * Incr double linefill disable (bit23=0)
+ * Prefetch offset = 7 (bit4:0)
+ */
+#define PL310_PREFETCH_CTRL_INIT	0x31000007
+
+/*
+ * PL310 Power Register
+ *
+ * Dynamic clock gating enabled
+ * Standby mode enabled
+ */
+#define PL310_POWER_CTRL_INIT		0x00000003
+
+/*
+ * SCU Invalidate Register
+ *
+ * Invalidate all registers
+ */
+#define	SCU_INV_CTRL_INIT		0xFFFFFFFF
+
+/*
+ * SCU Access Register
+ * - both secure CPU access SCU
+ */
+#define SCU_SAC_CTRL_INIT		0x0000000F
+
+/*
+ * SCU NonSecure Access Register
+ * - both nonsec cpu access SCU, private and global timer
+ */
+#define SCU_NSAC_CTRL_INIT		0x00000FFF
+
+/* define the memory areas */
+
+#ifdef CFG_WITH_PAGER
+
+/*
+ * TEE/TZ RAM layout:
+ *
+ *  +---------------------------------------+  <- CFG_CORE_TZSRAM_EMUL_START
+ *  | TEE private highly | TEE_RAM          |   ^
+ *  |   secure memory    |                  |   | CFG_CORE_TZSRAM_EMUL_SIZE
+ *  +---------------------------------------+   v
+ *
+ *  +---------------------------------------+  <- CFG_DDR_TEETZ_RESERVED_START
+ *  | TEE private secure |  TA_RAM          |   ^
+ *  |   external memory  |                  |   |
+ *  +---------------------------------------+   | CFG_DDR_TEETZ_RESERVED_SIZE
+ *  |     Non secure     |  SHM             |   |
+ *  |   shared memory    |                  |   |
+ *  +---------------------------------------+   v
+ *
+ *  TEE_RAM : default 256kByte
+ *  TA_RAM  : all what is left in DDR TEE reserved area
+ *  PUB_RAM : default 2MByte
+ */
+
+/* emulated SRAM, at start of secure DDR */
+
+#define CFG_CORE_TZSRAM_EMUL_START	0x4E000000
```

Comment:
should be `DRAM0_START + DRAM0_SIZE - 0x02000000`

Reply:
i did not want to change that in the first patch, but just reorganize the files.
i can make the changes in this patchset.
to tell the truth i am not even sure this part of the code still works.

Reply:
I do not want to break legacy in this patchset
This patchset is only used as a staring point for reorganisation of files.
i have another pull request with other patches to break legacy.
in that context i am little reluctant to even change the CFG_CORE_TZSRAM_EMUL_START	and related changes you have mentioned in this pull request.
let me know what you think

Reply:
The legacy for mx6q (imx6q/d/dl/s) is hard coded values for TZSRAM and TZDRAM:

no-pager: `TZDRAM_BASE/_SIZE = 0x4e000000/0x01f00000`
pager: `TZSRAM_BASE/_SIZE = 0x4e000000/0x00100000`
 .......... `TZDRAM_BASE/_SIZE = 0x4e100000/0x01e00000`
and both: `CFG_SHMEM_START/_SIZE = 0x4ff00000/0x00100000`

I would suggest to simply use these as:
```c
#define CFG_SHMEM_START	0x4ff00000
#define CFG_SHMEM_SIZE	0x00100000
#ifdef CFG_WITH_PAGER
#define TZSRAM_BASE	0x4e000000
#define TZSRAM_SIZE	CFG_CORE_TZSRAM_EMUL_SIZE
#define TZDRAM_BASE	(TZSRAM_BASE + CFG_TEE_RAM_VA_SIZE)
#define TZDRAM_SIZE	(0x01e00000 - CFG_TEE_RAM_VA_SIZE)
#else
#define TZDRAM_BASE	0x4e000000
#define TZDRAM_SIZE	0x01f00000
#endif
```

and replace `CFG_CORE_TZSRAM_EMUL_START`, `CFG_DDR_TEETZ_RESERVED_START/_SIZE` in the ascci-art figure above with the hardcode values.

Reply:
can we please address this as a separate pull request?
actually the hardcoded value 4e00000 is actually only valid for the sabredsd board (that have 1GByte of DDR) with sabre auto that has 2GByte this TZSRAM will endup at the middle of the DDR  and this is something that we do not want.

Reply:
This **imx6q.h** is applied to platforms `imx-mx6qsabrelite`, `imx-mx6qsabresd` and `imx-mx6dlsabresd`. The 2 first are not maintained and the later has no maintainer. @jforissier, maybe some update is interesting here. It looks like we can break legacy on these platforms.

I understand you want more flexibility. Do this in 2 steps. First rework the file layout, with hardcode addresses and sizes. Then, with another commit, let's introduce a `CFG_TZDRAM_BASE/_SIZE` for the imx platforms that wishes it (all if I understood well).

Reply:
@etienne-lms @jforissier , We (NXP) could take maintainer ship of thos platforms.

@etienne-lms i will wait for you other pull request before doing any more changes..

Reply:
@jforissier Is it ok to use SoC maintainer, but not board maintainer?